### PR TITLE
don't send empty sub list

### PR DESCRIPTION
**Pull request by johnzeng**
fix the issue of #496

### CODE REVIEW

**Comment by mcollina** [comment]
Can you add a unit test for this?

**Comment by johnzeng** [comment]
yes, let me check about it

**Comment by johnzeng** [comment]
should return an error (via callbacks) for empty topic list

File: test/abstract_client.js [hunk 1]
```diff
@@ -343,6 +343,17 @@ module.exports = function (server, config) {
       })
     })
 
+    it('should return an error (via callbacks) for empty topic list', function (done) {
+      var client = connect()
+      client.subscribe([], function (err) {
```
Comment (mcollina) [inline test/abstract_client.js]: we return an `Error` here.

Comment (johnzeng) [inline test/abstract_client.js]: ok.

Comment (anhldbk) [inline test/abstract_client.js]: @mcollina His test is ok, right?

**Pull request merged**